### ISSUE TITLE
Feature Requests:  Runtime configuration and plugins

### ISSUE THREAD

**Issue by stanier** [black-screen/black-screen#133]
I just have a couple of ideas I want to through your way, not sure if they're any good.

* Runtime configuration support for black-screen
* Support for runtime configurations of other shells (.bashrc, .zshrc)
* Plugins for black-screen.  Could possibly be called by run-time configuration.

**Comment by shockone** [comment]
@stanier, thank you for sharing! We do plan to have the last two, but I'm not sure I understand what the first one means. Could you elaborate, please?

**Comment by stanier** [comment]
I apologize, I should have tried to be more clear with it.

What I mean with the first point is that we should have a separate runtime configuration file (*.blackrc* or something similar) that defines instructions specifically for black-screen.  The reasoning for this is that while supporting runtime configuration for other shells is good for compatibility, it shouldn't be used to store runtime configuration for black-screen itself, as those other shells may not support configurations that black-screen does.

Another reasoning I had for this was more of an experimental concept: JavaScript-based rc.  There are many arguments for this, including that of the potential to let runtime configurations extend/modify the Electron shell.  It would also allow for data-related features that cannot be matched in bash or other shell languages (arrays and mapping would be available).

TL;DR To give a stable configuration option meant only for black-screen, as well as *possibly* provide the ability to run JavaScript in the runtime configuration.

**Comment by shockone** [comment]
@stanier, I should have guessed that. Any specific options you'd like to have?
